gtsfm/scene_optimizer.py

```diff
@@ -119,7 +178,26 @@ def create_computation_graph(
             exact_intrinsics,
         )
 
-        return i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph
+        # if we have the expected data, evaluate the computed relative pose
+        if i2Ti1_expected_graph is not None:
+            error_graphs = dask.delayed(self.__compute_metrics)(
+                i2Ri1_graph,
+                i2Ui1_graph,
+                dask.delayed(lambda x: x.rotation())(i2Ti1_expected_graph),
+                dask.delayed(lambda x: Unit3(x.translation()))(
+                    i2Ti1_expected_graph
+                ),
```

Comment:
I think it might be cleaner to move the logic to separate translation and rotation to the compute_metrics function. Also, doesnt the translation have to be normalized?

Reply:
Thanks. I casted it to Unit3 for normalization.

Reply:
Okay, you could also consider doing that in compute_metrics. What are your thoughts on that?

Reply:
Basically, passing ground truth Pose3 to _compute_metrics and allow it to separate them and normalize. Keeps code here cleaner.

Reply:
Thanks. implemented this